src/annotation.js

```diff
@@ -104,6 +113,9 @@ function updateElements(chart, state, options, mode) {
 		if (!el || !(el instanceof elType)) {
 			el = elements[i] = new elType();
 		}
+		const display = typeof annotation.display === 'function' ? callCallback(annotation.display, [chart, annotation], this) : valueOrDefault(annotation.display, true);
```

Comment:
I'm working on getting core handle these kinds of things, so plugin development is simpler and the interface would be consistent between plugins etc. But its work in progress so this is fine for now.

Reply:
@kurkle I have some doubts about the following code, to compute the display options (annotation.,js):

```javascript
const display = typeof annotation.display === 'function' ? callCallback(annotation.display, [chart, annotation], this) : valueOrDefault(annotation.display, true);
el._display = typeof display === 'boolean' ? display : false;
```

I don't know if I could write it better. Maybe there are some other better ways to do it. Otherwise, that's fine!